### DISCUSSION TITLE
Modelling relational events with the process of nodes merge/collapse throughout time?

### DISCUSSION THREAD

**Discussion by SamALIENWARE** [stocnet/goldfish#28] [Q&A]
In the merger and acquisition action, sometimes the buyer would acquire the majority stake of the target, and thus the two companies would become one. On networks, this is the process of node merge or node collapse, during which the ties of the target is "inherited" by the buyer. It is an interesting theoretical model but I do wonder how to actually fit it in the DyNAM framework. Is it possible for DyNAM to actually model the process of nodes "dying"? And even the process of "inheriting" (future alliance actions influenced by the action of one firm acquiring another)?

**Comment by auzaheta** [comment]
This is a fascinating problem. In `goldfish`, you can define when an actor joins or leaves the process. The present variable in the nodes data frame control this. You can create an event data frame with the dates when the actor is present or leaves, changing the value of present to TRUE or FALSE, respectively.  `Fisheries_Treaties_6070` has an example of how specific the changes. `sovchanges` data frame contains the dates when states join or leave the process.

```r
data("Fisheries_Treaties_6070")
states <- defineNodes(states)
states <- linkEvents(states, sovchanges, attribute = "present")
```

Now, the inheriting issue might not have a straightforward solution. I think one way to deal with it is by defining a third actor that represents the inheritance. Then, you define when the third actor joins and the other two leave the process. You might duplicate the events of the parent companies to the merged company to make the previous history match. I hope it makes sense.  @jhollway  and @stadtfeldethz  might have additional comments and insights on this.

**Reply by SamALIENWARE** [reply]
Thank you for providing the code and your suggestions. Appreciate it.

**Comment by jhollway** [comment]
Thanks @SamALIENWARE for raising this. The short answer is that there is no way to explicitly model node entry and/or collapse in DyNAMs at present. DyNAMs are a network event model with a primary focus on tie creation or deletion events. What can be done in `{goldfish}` at the moment is define when an actor joins or leaves the network, as @auzaheta said, so that the model takes into account which nodes are available to be chosen (either by the rate or by others in the choice functions).

In terms of the inheritance workaround that @auzaheta outlined, I might suggest creating the data such that the acquiring firm continues beyond the acquisition and only the acquired firm leaves the network, but perhaps take that date to update the acquiring firm's attributes (e.g. perhaps it grows in size?). Ultimately, the utility of this will all depend on what your network tie of interest is.

**Reply by SamALIENWARE** [reply]
Thank you @jhollway for your answers. The ties of interest are alliance ties, defined by some sort of inter-organizational collaboration. Combining your suggestion and @auzaheta 's : what if I specify the leave of the target (the acquired firm) and also assign the relational event history (previous ties) to the buyer (the acquiring firm)? The rationale is that: if events (ties) could be inherited, previous structural properties would be an "asset" for the buyer as well. This is impractical of course if DyNAMs would not produce robust results after such manipulation.

**Reply by jhollway** [reply]
Hmm, I'd probably need to think about this a little more, but what you could try is adding these "inherited" ties to the event list that updates the process state but not to the dependent event list (since these are inherited, not explicitly chosen).

(Undirected) alliance ties can be treated with the coordination DyNAM model (as outlined in [Stadtfeld, Hollway and Block 2017](https://journals.sagepub.com/doi/abs/10.1177/0081175017709295)).
